include/core/shape.h

```diff
@@ -1002,12 +1016,27 @@ class CompositeShape : public Shape<dim>
    * @brief Sets the proper dof handler, then computes/updates the map of cells
    * and their likely non-null nodes
    * @param updated_dof_handler the reference to the new dof_handler
-   * @param levels_not_precalculated the number of finer levels that won't be
-   * precalculated
+   * @param mesh_based_precalculations mesh-based precalculations that can lead to slight shape misrepresentation (if RBF typed)
    */
   void
-  update_precalculations(DoFHandler<dim> &  updated_dof_handler,
-                         const unsigned int levels_not_precalculated);
+  update_precalculations(DoFHandler<dim> &updated_dof_handler,
+                         const bool       mesh_based_precalculations);
+
+  /**
+   * @brief Load data from file. To be called at initialization, after repartitioning or when shape has moved.
+   * This function is used only for RBF shapes and its composites at the moment.
+   */
+  void
+  load_data_from_file();
+
+  /**
+   * @brief Remove data that doesn't affect the cells owned by local process   *
```

Comment:
```suggestion
   * @brief Remove data that doesn't affect the cells owned by local process 
```

What are cells owned? Do you mean not locally_owned and not ghost? artificial? be a bit clearer